homeassistant/components/lcn/config_flow.py

```diff
@@ -61,43 +80,43 @@ async def validate_connection(host_name: str, data: ConfigType) -> ConfigType:
         host, port, username, password, settings=settings
     )
 
-    await connection.async_connect(timeout=5)
+    try:
+        await connection.async_connect(timeout=5)
+        _LOGGER.debug("LCN connection validated")
+    except pypck.connection.PchkAuthenticationError:
+        _LOGGER.warning('Authentication on PCHK "%s" failed', host_name)
+        error = "authentication_error"
+    except pypck.connection.PchkLicenseError:
+        _LOGGER.warning(
+            'Maximum number of connections on PCHK "%s" was '
+            "reached. An additional license key is required",
+            host_name,
+        )
+        error = "license_error"
+    except (TimeoutError, ConnectionRefusedError):
+        _LOGGER.warning('Connection to PCHK "%s" failed', host_name)
+        error = "connection_refused"
 
-    _LOGGER.debug("LCN connection validated")
     await connection.async_close()
-    return data
+    return error
 
 
-class LcnFlowHandler(ConfigFlow, domain=DOMAIN):
+class LcnFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
     """Handle a LCN config flow."""
 
     VERSION = 1
 
-    async def async_step_import(self, data: ConfigType) -> ConfigFlowResult:
+    async def async_step_import(
+        self, data: ConfigType
+    ) -> config_entries.ConfigFlowResult:
         """Import existing configuration from LCN."""
-        host_name = data[CONF_HOST]
         # validate the imported connection parameters
-        try:
-            await validate_connection(host_name, data)
-        except pypck.connection.PchkAuthenticationError:
-            _LOGGER.warning('Authentication on PCHK "%s" failed', host_name)
-            return self.async_abort(reason="authentication_error")
-        except pypck.connection.PchkLicenseError:
-            _LOGGER.warning(
-                (
-                    'Maximum number of connections on PCHK "%s" was '
-                    "reached. An additional license key is required"
-                ),
-                host_name,
-            )
-            return self.async_abort(reason="license_error")
-        except TimeoutError:
-            _LOGGER.warning('Connection to PCHK "%s" failed', host_name)
-            return self.async_abort(reason="connection_timeout")
+        if (error := await validate_connection(data)) is not None:
+            return self.async_abort(reason=error)
```

Comment:
We need to tell the user the migration failed.

For reference, this is how it's done by the enigma2 integration https://github.com/home-assistant/core/blob/cf6588da07fafcd295ad457c77be207dfa3ea04c/homeassistant/components/enigma2/config_flow.py#L167-L197

Reply:
Added the issues and the respective entries in strings.json. Thanks for pointing me to the code snippet.